### PR TITLE
update phpdotenv

### PR DESCRIPTION
**Pull request by arosenhagen**
update vlucas/phpdotenv v3.1 and switch to new native constructor call

### CODE REVIEW

**Comment by shyim** [comment]
Would this not break the compability to the composer project? It requires version 2
https://github.com/shopware/composer-project/blob/master/composer.json#L23

**Comment by arosenhagen** [comment]
good catch - I made an [according PR](https://github.com/shopware/composer-project/pull/32) to that repo too. 👍

**Comment by shyim** [comment]
I don't really like to update that. If someone using still the old composer template, it would break on that installations. Maybe require version 2 or 3 and make a small version switch?

**Comment by shyim** [comment]
The template has been updated. Can you add a case to support v2 and v3?
This composer extension has still to support old composer-project templates :)

File: README.md [hunk 1]
```diff
@@ -31,7 +31,7 @@ Example composer.json
         "php": "^5.6.4||^7.0",
         "composer/installers": "^1.0",
         "shopware/shopware": "^5.4",
-        "vlucas/phpdotenv": "~2.0",
+        "vlucas/phpdotenv": "^3.1",
```
Comment (shyim) [inline README.md]: We need here a or condition, ~2.0 || ~3.1

**Review by shyim — CHANGES_REQUESTED** [review]

File: composer.json [hunk 2]
```diff
@@ -6,7 +6,7 @@
     "require": {
         "php": "^7.0.0",
         "composer-plugin-api": "^1.0.0",
-        "vlucas/phpdotenv": "~2.0",
+        "vlucas/phpdotenv": "^3.3",
```
Comment (shyim) [inline composer.json]: also here

**Comment by shyim** [comment]
Thanks! 💙

**Pull request merged**